perses/rjmc/topology_proposal.py

```diff
@@ -742,6 +745,181 @@ def propose(self):
         pass
 
 
+class SystemGenerator(object):
+    """
+    This is a utility class to generate OpenMM Systems from
+    topology objects.
+
+    Parameters
+    ----------
+    forcefields_to_use : list of string
+        List of the names of ffxml files that will be used in system creation.
+    metadata : dict, optional
+        Metadata associated with the SystemGenerator.
+    """
+
+    def __init__(self, forcefields_to_use, metadata=None):
+        self._forcefields = forcefields_to_use
+
+    def build_system(self, new_topology, forcefield_kwargs=None):
+        """
+        Build a system from the new_topology, adding templates
+        for the molecules in oemol_list
+
+        Parameters
+        ----------
+        new_topology : simtk.openmm.app.Topology object
+            The topology of the system
+        forcefield_kwargs : dict of arguments to createSystem, optional
+            Allows specification of various aspects of system creation.
+
+        Returns
+        -------
+        new_system : openmm.System
+            A system object generated from the topology
+        """
+        from openmoltools import forcefield_generators
+        forcefield = app.ForceField(*self._forcefields)
```

Comment:
You will want to move the creation of `forcefield` and the addition of the GAFF template generator into the constructor if you want `SystemGenerator` to cache new templates that are generated.